packages/cadl-ranch-specs/http/azure/client-generator-core/usage/main.tsp

```diff
@@ -55,6 +55,37 @@ namespace ModelInOperation {
   @route("/outputToInputOutput")
   @get
   op outputToInputOutput(): OutputModel;
+
+  model ResultModel {
+    name: string;
+  }
```

Comment:
Java should generate the class without constructor and setter.

```java
@Immutable
public final class ResultModel implements JsonSerializable<ResultModel> {
    /*
     * The name property.
     */
    @Generated
    private final String name;

    /**
     * Creates an instance of ResultModel class.
     * 
     * @param name the name value to set.
     */
    @Generated
    private ResultModel(String name) {
        this.name = name;
    }

    /**
     * Get the name property: The name property.
     * 
     * @return the name value.
     */
    @Generated
    public String getName() {
        return this.name;
    }
```

Reply:
I see, should we also consider other visibilities  on model type if that's the case? maybe no need? as we are not really support those visibilities for now?

Reply:
One case that occurs to me is the opposite

```ts
  @visibility("create", "update")
  prop: ModelOfInputUsage;
```

that `ModelOfInputUsage` has usage=input.
But at present, Java would produce same class of input and input+output.